### PR TITLE
refactor: use sinon to mock octokit methods

### PR DESCRIPTION
**Pull request by chingor13**
feat!: require node 18
feat!: remove `addOrUpdateIssueComment` which is moved to issue-utils.

### CODE REVIEW

File: packages/gcf-utils/src/configuration.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+export const RUNNING_IN_TEST = process.env.NODE_ENV === 'test';
+// Adding 30 second delay for each batch with 30 tasks
+export const DEFAULT_FLOW_CONTROL_DELAY_IN_SECOND = 30;
+
+export interface WrapConfig {
```
Comment (SurferJeffAtGoogle) [inline packages/gcf-utils/src/configuration.ts:20]: Please add a comment for this interface.

File: packages/gcf-utils/src/installations.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import {WrapConfig} from './configuration';
+
+export interface AppInstallation {
+  id: number;
+  targetType: string;
```
Comment (SurferJeffAtGoogle) [inline packages/gcf-utils/src/installations.ts:20]: Please add a comment for the targetType and login members.  I can guess what the other two mean.

**Review by SurferJeffAtGoogle — CHANGES_REQUESTED** [review]

File: packages/gcf-utils/src/configuration.ts (hunk 1 above)
Comment (chingor13) [reply packages/gcf-utils/src/configuration.ts:21]: Done

File: packages/gcf-utils/src/installations.ts (hunk 2 above)
Comment (chingor13) [reply packages/gcf-utils/src/installations.ts:23]: Done

**Review by SurferJeffAtGoogle — APPROVED** [review]

**Pull request merged**